aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -1018,6 +1019,38 @@ Tensor& heaviside_(Tensor& self, const Tensor& values) {
   return at::heaviside_out(self, self, values);
 }
 
+Tensor& ldexp_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  TORCH_CHECK(!self.is_complex(), "ldexp does not support complex tensors.");
+  TORCH_CHECK(isIntegralType(other.scalar_type()), "ldexp requires integral types for second argument.");
```

Comment:
This is an interesting limitation. It's consistent with Python, C++ and NumPy, however. Would you add a comment to noting that?

Nit: "ldexp requires an integral type for its second argument, but got <other.scalar_type()".